rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -2654,4 +2654,31 @@ rocalTranspose(
         ERR(e.what())
     }
     return output;
-}
\ No newline at end of file
+}
+
+RocalTensor rocalLog1p(RocalContext p_context,
+                             RocalTensor p_input,
+                             bool is_output) {
+    Tensor* output = nullptr;
+    if ((p_context == nullptr) || (p_input == nullptr)) {
+        ERR("Invalid ROCAL context or invalid input tensor")
+        return output;
+    }
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    try {
+        RocalTensorDataType op_tensor_data_type = static_cast<RocalTensorDataType>(input->data_type());
+        if (op_tensor_data_type != RocalTensorDataType::INT16) {
+            THROW("Log1p augmentation only supported for int16 inputs")
+        }
+        op_tensor_data_type = RocalTensorDataType::FP32;
```

Comment:
Says supported only for INT16 and reassigns to FP32. I am confused

Reply:
op_tensor_data_type is output_tensor_data_type. Log1p rpp kernel only supports i16->f32

Reply:
please add a comment for this. Since you are using the same variable, it's confusing